src/utils/constants.js

```diff
@@ -65,4 +65,6 @@ export const TOOLTIP_DESCRIPTIONS = {
   propertyNames: 'Names of properties must follow a specified convention.',
   patternProperties:
     'Property names or values should match the specified pattern.',
+  required: 'Required property',
+  contains: 'Only needs to validate against one or more items in the array.',

```

Comment:
If the tooltip fits in one sentence, let's default to no period at the end.

Reply:
Applied changes!